### PR TITLE
mongodb: 4.0.12 -> 4.2.8

### PR DESCRIPTION
**Pull request by pjjw**
Not strictly an upgrade, but adds a new mongodb-4_2 target with the
current mongodb from that branch.

###### Motivation for this change
Current mongodb isn't packaged, would like it to be.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/cheetah3/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "ececc9ca7c58b9a86ce71eb95594c4619949e2a058d2a1af74c7ae8222515eb1";
+  };
+
+  doCheck = false; # Circular dependency
+
+  meta = {
+    homepage = http://www.cheetahtemplate.org/;
```
Comment (fgaz) [inline pkgs/development/python-modules/cheetah3/default.nix:15]: Quotes around the url https://github.com/NixOS/rfcs/pull/45

File: pkgs/development/python-modules/cheetah3/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    description = "A template engine and code generation tool";
+    license = lib.licenses.mit;
+  };
+}
```
Comment (fgaz) [inline pkgs/development/python-modules/cheetah3/default.nix:19]: Nixpkgs' style guide requires a newline at eof

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 3]
```diff
@@ -11,6 +11,17 @@ with stdenv.lib;
 }@args:
 
 let
+  ifdef = if versionAtLeast version "4.2"
```
Comment (fgaz) [inline pkgs/servers/nosql/mongodb/mongodb.nix:14]: "ifdef" is not very self-explanatory, I'd change it with something like "pythonVariant"

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 4]
```diff
@@ -11,6 +11,17 @@ with stdenv.lib;
...
+    then { python = python38.withPackages (ps: with ps; [ pyyaml cheetah3 psutil setuptools ]);
+            scons = scons;
```
Comment (fgaz) [inline pkgs/servers/nosql/mongodb/mongodb.nix:16]: Maybe inherit? I do like the symmetry with the one below though :)

File: pkgs/development/python-modules/cheetah3/default.nix [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    sha256 = "ececc9ca7c58b9a86ce71eb95594c4619949e2a058d2a1af74c7ae8222515eb1";
+  };
+
+  doCheck = false; # Circular dependency
+
+  meta = {
+    homepage = http://www.cheetahtemplate.org/;
+    description = "A template engine and code generation tool";
+    license = lib.licenses.mit;
+  };
```
Comment (fgaz) [inline pkgs/development/python-modules/cheetah3/default.nix:18]: What about adding yourself as the maintainer of the library? :)

**Review by fgaz — APPROVED** [review]
The diff looks fine (didn't look at the `.patch`es). Some style suggestions.

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 3 above)
Comment (pjjw) [reply pkgs/servers/nosql/mongodb/mongodb.nix:14]: yeah, good suggestion, will do, thanks!

File: pkgs/development/python-modules/cheetah3/default.nix (hunk 2 above)
Comment (pjjw) [reply pkgs/development/python-modules/cheetah3/default.nix:19]: done

File: pkgs/development/python-modules/cheetah3/default.nix (hunk 1 above)
Comment (pjjw) [reply pkgs/development/python-modules/cheetah3/default.nix:15]: done

File: pkgs/development/python-modules/cheetah3/default.nix (hunk 5 above)
Comment (pjjw) [reply pkgs/development/python-modules/cheetah3/default.nix:18]: done

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 4 above)
Comment (pjjw) [reply pkgs/servers/nosql/mongodb/mongodb.nix:16]: this reads a bit clearer to me, though admittedly i am a novice here.

**Comment by pjjw** [comment]
addressed comments, squashed out the errors i'd made, noticed that github doesn't preserve that history- apologies for that. regardless, PTAL. thanks!

**Comment by fgaz** [comment]
@GrahamcOfBorg build mongodb-4_2

**Comment by fgaz** [comment]
Oh, right. It's unfree now.

**Comment by pjjw** [comment]
actually just caught an issue with tests- current test is using cross-version clients, and 3.4 to 4.2 doesn't work. quick update inbound.

**Review by fgaz — APPROVED** [review]
I pinged a few more people that were in the maintainer list. If no one replies in a couple days I'll just set this to needs_merge

**Comment by fgaz** [comment]
In the meantime please prefix the commit messages with their scope, like you did on the first one

**Comment by pjjw** [comment]
sure. mind if i just squash them out now that you've reviewed?

**Comment by fgaz** [comment]
That's also fine, yes

**Review by phile314 — APPROVED** [review]

**Review by offlinehacker — APPROVED** [review]

**Pull request merged**